include/dlr_common.h

```diff
@@ -125,12 +127,12 @@ class DLR_DLL DLRModel {
 
   /* Output related functions */
   virtual int GetNumOutputs() { return num_outputs_; }
-  virtual const char* GetOutputName(const int index) const { 
-    LOG(ERROR) << "GetOutputName is not supported yet!";
-    return NULL;
+  virtual const char* GetOutputName(const int index) const {
+    throw dmlc::Error("GetOutputName is not supported for this model.");
+    return nullptr;
   }
-  virtual int GetOutputIndex(const char* name) const { 
-    LOG(ERROR) << "GetOutputName is not supported yet!";
+  virtual int GetOutputIndex(const char* name) const {
+    throw dmlc::Error("GetOutputIndex is not supported for this model.");
```

Comment:
👍

Reply:
Are we catching these errors in dlr.cc (C-APIs)? We don't have very many negative tests for C-APIs so not sure if this will crash on bad call.

Reply:
Yeah, the `API_BEGIN` and `API_END` macros will add a try/catch(dmlc::Error) to all of the C functions

Reply:
oh nice. I didn't know that. I manually added them in few places. Good to know. Thanks